cupy/core/core.pyx

```diff
@@ -2833,19 +2832,30 @@ cpdef ndarray tensordot_core(
         c = c.view()
         c.shape = (n, m)
 
-    if not use_sgemmEx:
-        a = a.astype(dtype, copy=False)
-        b = b.astype(dtype, copy=False)
+    a = a.astype(dtype, copy=False)
+    b = b.astype(dtype, copy=False)
 
     # Be careful that cuBLAS uses the FORTRAN-order matrix representation.
-    handle = device.get_cublas_handle()
     # Matrix-Matrix product A^T * B
     # c is C-contiguous while cuBLAS assumes F-contiguous inputs, so we
     # compute C^T = B^T * A here.
     a, transa, lda = _mat_to_cublas_contiguous(a, 0)
     b, transb, ldb = _mat_to_cublas_contiguous(b, 1)
-    if use_sgemmEx:
-        Ctype = runtime.CUDA_R_16F if c.dtype == 'e' else runtime.CUDA_R_32F
+
+    global _cuda_runtime_version
```

Comment:
Note: I think we no longer need to check this as we're on 9.0+ starting CUDA v8! I will send a PR to remove it from a few places, but for the ease of backport let's keep it here.